Sources/SwiftRefactor/ExpandEditorPlaceholder.swift

```diff
@@ -94,18 +92,18 @@ struct ExpandSingleEditorPlaceholder: EditRefactoringProvider {
 
     let expanded: String
     if let functionType = placeholder.typeForExpansion?.as(FunctionTypeSyntax.self) {
-      let basicFormat = BasicFormat(
+      let format = ClosureLiteralFormat(
         indentationWidth: context.indentationWidth,
         initialIndentation: context.initialIndentation
       )
-      var formattedExpansion = functionType.closureExpansion.formatted(using: basicFormat).description
+      var formattedExpansion = functionType.closureExpansion.formatted(using: format).description
       // Strip the initial indentation from the placeholder itself. We only introduced the initial indentation to
       // format consecutive lines. We don't want it at the front of the initial line because it replaces an expression
       // that might be in the middle of a line.
       if formattedExpansion.hasPrefix(context.initialIndentation.description) {
         formattedExpansion = String(formattedExpansion.dropFirst(context.initialIndentation.description.count))
       }
-      expanded = formattedExpansion
+      expanded = wrapInPlaceholder(formattedExpansion)
```

Comment:
ℹ️ I think wrapping the whole closure in a placeholder, with inner placeholders for the args and body, provides an optimal UX: you can immediately delete/replace the entire thing or you can tab to the constituent parts. This does necessitate a change in sourcekit-lsp's placeholder translation routine to handle the nesting; this is done in https://github.com/swiftlang/sourcekit-lsp/pull/1831

Reply:
Unfortunately, we can’t do that. The Swift compiler and Xcode don’t support nested editor placeholders using the `<# … #>` syntax. But I do agree that wrapping the closure in a placeholder again makes sense in the context of automatic SourceKit-LSP expansions. Do you think we could do the re-wrapping on the SourceKit-LSP side?

Reply:
Darn, I was afraid that might be the case. Okay, following this and your other comments, I'll figure out how to move the bulk of the behavior change into sourcekit-lsp.

Reply:
I don't see a (good) way to move the nested wrapping into SourceKit-LSP; we'd essentially have to re-do most of this work: re-parsing the expansion and finding the closure body to wrap. I have added a flag to the `Context` to control this. It is off by default and `CodeCompletionSession` will set it for its own expansion requests.